src/sentry/snuba/metrics/query_builder.py

```diff
@@ -925,6 +925,37 @@ def _build_orderby(self) -> Optional[List[OrderBy]]:
 
         return orderby_fields
 
+    def _build_having(self) -> List[Union[BooleanCondition, Condition]]:
```

Comment:
Nice catch!

The thing is that, do we want to support all derived metrics in the having?

Reply:
I don't really want to support HAVING at all, but I don't see a reason to restrict the derived metrics.

Reply:
We actually backtracked on the decision to support having and wanted to delete it. There was a miscommunication between teams that requested us to implement it. If I am correct, it's not used.

Reply:
It is used, certainly in a number of tests. Hence why I have this PR. You can see which tests here: https://github.com/getsentry/sentry/pull/55080/files#diff-126943646d55760ffab2a29b719f2a6b0cf2d14c5a65ff1c65fb46e7561f5e42L2460